robot/impl/local_robot.go

```diff
@@ -623,6 +623,12 @@ func (r *localRobot) newResource(
 		}
 	}()
 	resName := conf.ResourceName()
+
+	// create logger to make sure errors can be logged before doing anything
+	resLogger := r.logger.Sublogger(resName.String())
+	resLogger.SetLevel(conf.LogConfiguration.Level)
+	gNode.SetLogger(resLogger)
```

Comment:
This change is subtle, the relationship between a gNode and its logger is hidden across these files. If I tracked the code correctly, we now [ensure this](https://github.com/viamrobotics/rdk/blob/2a1ed09fe7164ca92cc0717d0d3100ebcdd1cc02/robot/impl/resource_manager.go#L615-L621) `gNode.LogAndSetError` gets hit?

So if I understood correctly, it seems plausible [that these](https://github.com/viamrobotics/rdk/blob/2a1ed09fe7164ca92cc0717d0d3100ebcdd1cc02/robot/impl/resource_manager.go#L585-L601) `LogAndSetError` calls can still be missed (I think this `gNode.SetLogger` gets [set on](https://github.com/viamrobotics/rdk/blob/2a1ed09fe7164ca92cc0717d0d3100ebcdd1cc02/robot/impl/resource_manager.go#L605) `processResource`).

If what I said is true, can you take a minute and see if it makes sense to initialize the logger sooner? If it's troublesome -- no need to worry. What you have is a definitive improvement and I don't want to slow down what you're actually trying to do!